datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1097,13 +1098,15 @@ private void validateSdxResizeRequest(SdxCluster sdxCluster, String accountId, S
         }
     }
 
-    private void validateRuntimeAndImage(SdxClusterRequest clusterRequest, DetailedEnvironmentResponse environment,
+    @VisibleForTesting
+    void validateRuntimeAndImage(SdxClusterRequest clusterRequest, DetailedEnvironmentResponse environment,
             ImageSettingsV4Request imageSettingsV4Request, ImageV4Response imageV4Response) {
         ValidationResultBuilder validationBuilder = new ValidationResultBuilder();
         CloudPlatform cloudPlatform = EnumUtils.getEnumIgnoreCase(CloudPlatform.class, environment.getCloudPlatform());
 
         if (imageV4Response != null) {
-            if (clusterRequest.getRuntime() != null && !Objects.equals(clusterRequest.getRuntime(), imageV4Response.getVersion())) {
+            if (imageV4Response.getVersion() != null && clusterRequest.getRuntime() != null
```

Comment:
I would use `StringUtils.isNotBlank` instead of the null check and also cover the empty string case with an UT